CONTRIBUTING.md

```diff
@@ -1,22 +1,78 @@
+If you are here, it means you are interested in helping us out. A hearty welcome and thank you! There are many ways you can contribute to the Hyperspace project:	
 
 * Offer PR's to fix bugs or implement new features.
-* Give us feedback and bug reports regarding the software or the documentation.
-* Improve our examples, tutorials, and documentation.
+* Review currently [open PRs](https://github.com/microsoft/hyperspace/pulls)	
+* Give us feedback and bug reports regarding the software or the documentation.	
+* Improve our design docs, examples, tutorials, and documentation.
 
 Please start by browsing the [issues](https://github.com/microsoft/hyperspace/issues) and leave a comment to engage us if any of them interests you.
 
-Please make sure to take a look at the project [roadmap](ROADMAP.md).
+Please make sure to take a look at the project [roadmap](ROADMAP.md).	
 
-## Contributing Code
+# Contributing Code
 
-* Before starting working on a major feature or bug fix, please open a GitHub issue describing the work you are proposing. We will make sure no one else is already working on it and the work aligns with the project [roadmap](ROADMAP.md).
-* A "major" feature or bug fix is defined as any change that is > 100 lines of code (not including tests) or changes user-facing behavior (e.g., breaking API changes).
-* Once you are ready, you can create a [PR](https://github.com/microsoft/hyperspace/pulls) and the committers will help reviewing your PR.
+* Before starting working on a major feature or bug fix, please open a GitHub issue describing the work you are proposing. We will make sure no one else is already working on it and the work aligns with the project [roadmap](ROADMAP.md).	
+* A "major" feature or bug fix is defined as any change that is > 100 lines of code (not including tests) or changes user-facing behavior (e.g., breaking API changes). Please read [Proposing Major Changes to Hyperspace](#proposing-major-changes-to-hyperspace) before you begin any major work.	
+* Once you are ready, you can create a [PR](https://github.com/microsoft/hyperspace/pulls) and the committers will help reviewing your PR.	
 
+**Coding Style: ** Please review our [coding guidelines](/docs/coding-guidelines/scala-coding-style.md).	
 
-### Coding Style
+# Proposing Major Changes to Hyperspace
 
-Please review our [coding guidelines](/docs/coding-guidelines/scala-coding-style.md).
+## Introduction	
+
+The development process in Hyperspace is design-driven. If you intend of making any significant changes, please consider discussing with the Hyperspace community first (and sometimes formally documented), before you open a PR.
+
+The rest of this document describes the process for proposing, documenting and implementing changes to the Hyperspace project.
+
+To learn about the motivation behind Hyperspace, see the talk [Hyperspace: An Indexing Subsystem for Apache Spark](https://www.youtube.com/watch?v=ofn53mT7H6c) from Spark+AI Summit 2020.	
```

Comment:
Lots of weird whitespaces at the ends of lines.  Not sure what's causing these.

Reply:
Thanks! Resolved it.